### PR TITLE
Tau dqm aod to miniaod for mc

### PR DESCRIPTION
**Pull request by easilar**
#### PR description:

In broad strokes, the goal of this reorganization is to put standardValidation (which takes AOD inputs) and miniaodValidation (MiniAOD inputs) on equal footing. In doing so, more histograms will be created in the miniaodValidation step, and the standardValidation procedure will become obsolete. 
 -First Plan presentation in DQM general meeting: https://indico.cern.ch/event/870363/contributions/3670599/attachments/1975330/3287515/Tau_DQM_SequenceReview_Jan21st.pdf
  -The details of the planned changes are presented in Tau POG meetings and in PPD meeting: https://indico.cern.ch/event/997168/contributions/4288672/attachments/2214999/3749685/PPDWorkshop.pdf
 -Related twiki page: https://twiki.cern.ch/twiki/bin/view/CMS/SWGuidePFTauIDValidation

#### PR validation:

PR passes the basic test procedure :
```
cd src
scram b distclean 
git cms-checkdeps -a -A
scram b -j 8
scram b runtests
```

PR passes also additional tests for tau dqm offline validation:
```
 cd Validation/RecoTau/test/
 cmsRun config.py
 cmsRun harvesting.py
```

### CODE REVIEW

File: Validation/RecoTau/python/dataTypes/ValidateTausOnZTT_cff.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 from PhysicsTools.HepMCCandAlgos.genParticles_cfi import *
 import PhysicsTools.PatAlgos.tools.helpers as helpers
 
-tauGenJetsForVal = tauGenJets.clone()
+tauGenJetsForVal = tauGenJets.clone(GenParticles = cms.InputTag('prunedGenParticles'))
```
Comment (jfernan2) [inline Validation/RecoTau/python/dataTypes/ValidateTausOnZTT_cff.py:8]: Please drop type, i.e.
tauGenJetsForVal = tauGenJets.clone(GenParticles = 'prunedGenParticles')
following point 17. on:
https://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1

File: Validation/RecoTau/src/TauTagValidation.cc [hunk 2]
```diff
@@ -206,6 +206,8 @@ void TauTagValidation::bookHistograms(DQMStore::IBooker& ibooker,
   int j = 0;
   for (const auto& it : discriminators_) {
     string DiscriminatorLabel = it.getParameter<string>("discriminator");
+    std::cout << "Current discriminator: \n";
```
Comment (jfernan2) [inline Validation/RecoTau/src/TauTagValidation.cc:209]: cout is not thread safe, please use edm::Log

File: Validation/RecoTau/src/TauValidationMiniAOD.cc [hunk 3]
```diff
@@ -60,107 +157,375 @@ void TauValidationMiniAOD::bookHistograms(DQMStore::IBooker& ibooker,
...
+    phiLoosevsMuoMap.insert(std::make_pair("",  phiLoosevsMuo));
+    massLoosevsMuoMap.insert(std::make_pair("", massLoosevsMuo));
+    puLoosevsMuoMap.insert(std::make_pair("",   puLoosevsMuo));
+  }
 }
 
 void TauValidationMiniAOD::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  
+  // temp
+  std::cout << "******** Entering the Analyze Function ********\n";
```
Comment (jfernan2) [inline Validation/RecoTau/src/TauValidationMiniAOD.cc:451]: cout is not thread safe, use edm::Log instead

File: Validation/RecoTau/python/RecoTauValidation_cff.py [hunk 4]
```diff
@@ -19,25 +26,25 @@
 )
 tauValidationMiniAODRealData = tauValidationMiniAODZTT.clone(
   RefCollection = cms.InputTag("CleanedPFJets"),
```
Comment (jfernan2) [inline Validation/RecoTau/python/RecoTauValidation_cff.py:28]: Please drop type, i.e.
tauValidationMiniAODZTT.clone(
  RefCollection = "CleanedPFJets",
ExtensionName = 'JETHT')
following point 17. on:
https://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1

File: Validation/RecoTau/python/RecoTauValidation_cff.py [hunk 5]
```diff
@@ -19,25 +26,25 @@
...
-  ExtensionName = cms.string('RealData')
+  ExtensionName = cms.string('JETHT')
 )
 tauValidationMiniAODRealElectronsData = tauValidationMiniAODZTT.clone(
   RefCollection = cms.InputTag("ElZLegs","theProbeLeg"),
```
Comment (jfernan2) [inline Validation/RecoTau/python/RecoTauValidation_cff.py:32]: same comment as above

File: Validation/RecoTau/python/RecoTauValidation_cff.py [hunk 6]
```diff
@@ -19,25 +26,25 @@
...
-  ExtensionName = cms.string("RealElectronsData")
+  ExtensionName = cms.string("DoubleElectron")
 )
 tauValidationMiniAODRealMuonsData = tauValidationMiniAODZTT.clone(
   RefCollection = cms.InputTag("MuZLegs","theProbeLeg"),
```
Comment (jfernan2) [inline Validation/RecoTau/python/RecoTauValidation_cff.py:36]: same as above

File: Validation/RecoTau/python/RecoTauValidation_cff.py (hunk 4 above)
Comment (easilar) [reply Validation/RecoTau/python/RecoTauValidation_cff.py:28]: Hello, these definitions are coming from already previously used releases and also in the CMSSW master branch [1]. Should we edit all the cloned types in this file ?
 
[1] https://github.com/cms-sw/cmssw/blob/master/Validation/RecoTau/python/RecoTauValidation_cff.py#L8-L31

Comment (jfernan2) [reply Validation/RecoTau/python/RecoTauValidation_cff.py:28]: If it does not cost you much, yes please!

Comment (jfernan2) [reply Validation/RecoTau/python/RecoTauValidation_cff.py:28]: Please bear in mind that these changes about drop type have been just implemented in this PR from the DQM central campaign:
https://github.com/cms-sw/cmssw/pull/36511

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
please test

**Comment by easilar** [comment]
Hello,
As I am not receiving any error in my local pipeline,  I am checking the errors provided here. But I am not able to see any details of the errors. 
Also the directory /cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/36557/ does not exist. 
Thank you, 
Ece

**Comment by jfernan2** [comment]
@easilar you should rebase your changes to CMSSW master version (currently 12_3_X) unless you have a strong reason to use 11_X, in which case please target first master and then justify the backport

A full check with runTheMatrix -l limited should be tested before submitting as explained here:
http://cms-sw.github.io/tutorial.html

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

Thanks

**Comment by perrotta** [comment]
-1
- This looks unattended, and in any case it cannot be submitted to 11_0_X
- @easilar please close this PR, and move it to master if you really plan to have it merged eventually

**Comment by easilar** [comment]
I am preparing a PR to master branch now. Thank you.

**Pull request closed without merging**